{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Building a concatenate species tree with ete build\n",
    "Besides the reconstruction of gene phylogenies, ETE build provides basic workflows to build concatenated trees. \n",
    "You only need to provide a file defining the target COGs to use. This is, a file where each lines contains a tab-delimited list of sequences that are considered orthologs among all taxa.\n",
    "\n",
    "- Relevant documentation: [ete3 build supermatrix](http://etetoolkit.org/cookbook/ete_build_supermatrix.ipynb)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Infer species tree based on a concatenated alignment of marker gene families"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a COGS file that ete build can read\n",
    "To play well with ETE, the COGs file should contain sequences names where the species code can be easily extracted. \n",
    "\n",
    "Remember that, in our case, all sequences have the following format:\n",
    "\n",
    "- `spcode.seqname`, where `.` acts as a delimiter."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load trees and marker trees\n",
    "import pickle\n",
    "\n",
    "def extract_spcode(nodename):\n",
    "    return nodename.split('.')[0]\n",
    "\n",
    "all_trees = pickle.load(open('data/alltrees.pkl', 'rb'))\n",
    "sptree_markers = pickle.load(open('data/sptree_markers.pkl', 'rb'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a COGS file that ete build can read"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TASK: Create a text file where each line contains a tab-delimited list of sequences of a marker gene family (COG)\n",
    "# Save the file as \"data/marker_cogs.tsv\" in your data folder\n",
    "\n",
    "with open('data/marker_cogs.tsv', 'w') as COGS:\n",
    "    for tname in sptree_markers: \n",
    "        t = all_trees[tname]\n",
    "        print('\\t'.join(t.get_leaf_names()), file=COGS)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "933801.Ahos_0739\t273063.STK_02680\t673860.AciM339_0977\t263820.PTO0416\t1051632.TPY_0322\t224324999.sul008\t224324.aq_008\t525897.Dbac_2775\t743299.Acife_2711\t637389.Acaty_c0617\t555778.Hneap_0322\t1255043.TVNIR_2575\t713587.THITH_14295\t1158165.KB898880_gene1474\t1121405.dsmv_3590\r\n",
      "273063.STK_04260\t933801.Ahos_0848\t673860.AciM339_1102\t263820.PTO0644\t1121405.dsmv_3585\t525897.Dbac_2770\t224324999.sul015\t224324.aq_015\t1051632.TPY_0327\t1158165.KB898880_gene1479\t1255043.TVNIR_2570\t713587.THITH_14270\t555778.Hneap_0327\t637389.Acaty_c0622\t743299.Acife_2706\r\n",
      "273063.STK_04220\t933801.Ahos_0854\t263820.PTO0650\t673860.AciM339_1108\t224324.aq_1654\t224324999.sul1654\t525897.Dbac_2764\t637389.Acaty_c0628\t743299.Acife_2700\t555778.Hneap_0333\t713587.THITH_14240\t1255043.TVNIR_2564\t1158165.KB898880_gene1485\t1121405.dsmv_3579\t1051632.TPY_0333\r\n",
      "933801.Ahos_0662\t273063.STK_20680\t673860.AciM339_1558\t263820.PTO1221\t555778.Hneap_0345\t637389.Acaty_c0643\t743299.Acife_2685\t1255043.TVNIR_2552\t713587.THITH_14175\t1158165.KB898880_gene1497\t525897.Dbac_2751\t224324999.sul073\t224324.aq_073\t1051632.TPY_0348\t1121405.dsmv_3566\r\n",
      "933801.Ahos_0664\t273063.STK_20700\t673860.AciM339_1560\t263820.PTO1219\t224324999.sul074\t224324.aq_074\t1051632.TPY_0347\t743299.Acife_2686\t637389.Acaty_c0642\t555778.Hneap_0344\t1255043.TVNIR_2553\t1158165.KB898880_gene1496\t713587.THITH_14180\t525897.Dbac_2752\t1121405.dsmv_3567\r\n",
      "933801.Ahos_0659\t273063.STK_20650\t673860.AciM339_1029\t263820.PTO0323\t525897.Dbac_0049\t1121405.dsmv_2426\t1051632.TPY_0368\t224324.aq_1877\t224324999.sul1877\t1158165.KB898875_gene68\t713587.THITH_03880\t1255043.TVNIR_3188\t555778.Hneap_2044\t743299.Acife_3087\t637389.Acaty_c0470\r\n",
      "224324999.sul1590\t224324.aq_1590\t1121405.dsmv_0203\t525897.Dbac_2890\t1158165.KB898872_gene767\t713587.THITH_11925\t1255043.TVNIR_0872\t555778.Hneap_2190\t743299.Acife_1511\t637389.Acaty_c1132\t263820.PTO1283\t673860.AciM339_1327\t1051632.TPY_1759\t933801.Ahos_0752\t273063.STK_02800\r\n",
      "273063.STK_13680\t933801.Ahos_1295\t263820.PTO0437\t673860.AciM339_0145\t224324.aq_1933\t224324999.sul1933\t1051632.TPY_0300\t743299.Acife_2721\t637389.Acaty_c0607\t1158165.KB898884_gene2556\t555778.Hneap_0312\t1255043.TVNIR_2585\t713587.THITH_14345\t1121405.dsmv_3649\t525897.Dbac_2785\r\n",
      "273063.STK_03940\t933801.Ahos_0726\t263820.PTO0714\t673860.AciM339_0319\t224324.aq_132\t224324999.sul132\t1051632.TPY_0230\t1121405.dsmv_1592\t525897.Dbac_0059\t743299.Acife_2731\t637389.Acaty_c0594\t555778.Hneap_1866\t1158165.KB898872_gene937\t1255043.TVNIR_0990\t713587.THITH_11630\r\n",
      "933801.Ahos_0520\t273063.STK_04720\t263820.PTO1061\t673860.AciM339_1250\t224324999.sul527\t224324.aq_527\t637389.Acaty_c2204\t743299.Acife_2485\t555778.Hneap_0590\t1158165.KB898871_gene2506\t713587.THITH_01500\t1255043.TVNIR_3592\t1051632.TPY_0961\t525897.Dbac_3477\t1121405.dsmv_2875\r\n"
     ]
    }
   ],
   "source": [
    "# Let's have a look at the COGs file\n",
    "!head data/marker_cogs.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use ete build to infer a supermatrix tree out of your marker families\n",
    "\n",
    "- specify a gene tree workflow (for the gene alignment)\n",
    "- specify a supermatrix workflow (how to concatenate COGs and program use to infer the final tree)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Toolchain path: /home/huerta/eteconda/envs/etecourse/bin/ete3_apps/bin \n",
      "Toolchain version: unknown\n",
      "['mafft_default-none-none-none']\n",
      "['cog_all-alg_concat_default-fasttree_full']\n",
      "\n",
      "      --------------------------------------------------------------------------------\n",
      "                  ETE build (3.1.2) - reproducible phylogenetic workflows\n",
      "\n",
      "      Citation:\n",
      "\n",
      "       Huerta-Cepas J, Serra F and Bork P. ETE 3: Reconstruction, analysis and\n",
      "       visualization of phylogenomic data. Mol Biol Evol (2016)\n",
      "       doi:10.1093/molbev/msw046\n",
      "\n",
      "      (Note that a list of the external programs used to complete all necessary\n",
      "      computations will be shown after workflow execution. Those programs should\n",
      "      also be cited.)\n",
      "\n",
      "      --------------------------------------------------------------------------------\n",
      "      \n",
      "\u001b[32mINFO\u001b[0m -  Testing x86-64  portable applications...\n",
      "       clustalo: \u001b[32mOK\u001b[0m - 1.2.4\n",
      "      dialigntx: \u001b[32mOK\u001b[0m - This is DIALIGN-TX Version 1.0.2 - A Multiple Sequence alignment program.\n",
      "       fasttree: \u001b[32mOK\u001b[0m - Usage for FastTree version 2.1.8 Double precision (No SSE3), OpenMP (2 threads):\n",
      "         iqtree: \u001b[32mOK\u001b[0m - IQ-TREE multicore version 1.5.5 for Linux 64-bit built Jun  2 2017\n",
      "         kalign: \u001b[32mOK\u001b[0m - Kalign version 2.04, Copyright (C) 2004, 2005, 2006 Timo Lassmann\n",
      "          mafft: \u001b[32mOK\u001b[0m - MAFFT v6.861b (2011/09/24)\n",
      "         muscle: \u001b[32mOK\u001b[0m - MUSCLE v3.8.31 by Robert C. Edgar\n",
      "          phyml: \u001b[32mOK\u001b[0m - . This is PhyML version 20160115.\n",
      "     pmodeltest: \u001b[32mOK\u001b[0m - pmodeltest.py v1.4\n",
      "          raxml: \u001b[32mOK\u001b[0m - This is RAxML version 8.2.11 released by Alexandros Stamatakis on June 2017.\n",
      " raxml-pthreads: \u001b[32mOK\u001b[0m - This is RAxML version 8.2.11 released by Alexandros Stamatakis on June 2017.\n",
      "         readal: \u001b[32mOK\u001b[0m - readAl v1.4.rev6 build[2012-02-02]\n",
      "         statal: \u001b[32mOK\u001b[0m - statAl v1.4.rev6 build[2012-02-02]\n",
      "        tcoffee: \u001b[31mMISSING\u001b[0m\n",
      "         trimal: \u001b[32mOK\u001b[0m - trimAl v1.4.rev6 build[2012-02-02]\n",
      "\u001b[1;33m\n",
      "WARNING: 1 external tools seem to be missing or unfunctional\u001b[0m\n",
      "\u001b[1;32mInstall using conda (recomended):\u001b[0m\n",
      "\u001b[0m conda install -c etetoolkit ete3_external_apps\u001b[0m\n",
      "\u001b[1;32mor manually compile by running:\u001b[0m\n",
      "\u001b[0m ete3 upgrade-external-tools\u001b[0m\n",
      "\n",
      "\u001b[32mINFO\u001b[0m -  Starting ETE-build execution at Tue Sep  1 12:41:55 2020\n",
      "\u001b[32mINFO\u001b[0m -  Output directory /home/huerta/etecourse/course/nb/03-building_concat_species_trees/data/sptree\n",
      "\u001b[32mINFO\u001b[0m -  Erasing all existing npr data...\n",
      "\u001b[1;37;43mWRNG\u001b[0m -  COG file restriction: 495 sequences from 15 species \n",
      "\u001b[32mINFO\u001b[0m -  Reading aa sequences from data/all_prots.faa...\n",
      "loaded:0000150 skipped:0009850 scanned:0010000 30.3%\n",
      "loaded:0000275 skipped:0019725 scanned:0020000 55.6%\n",
      "loaded:0000390 skipped:0029610 scanned:0030000 78.8%\n",
      "loaded:0000490 skipped:0039510 scanned:0040000 99.0%\n",
      "\n",
      "\n",
      "\u001b[1;37;43mWRNG\u001b[0m -  495 target sequences\n",
      "\u001b[32mINFO\u001b[0m -  ETE build starts now!\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:41:55 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[34mCogSelectorTask\u001b[0m (15 species, MCL-COGs, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 0 jobs. 0(R), 0(W). Cores usage: 0/5\n",
      "\u001b[32mINFO\u001b[0m -                     933801  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     273063  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     673860  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     263820  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                    1051632  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                  224324999  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     224324  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     525897  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     743299  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     637389  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     555778  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                    1255043  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     713587  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                    1158165  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                    1121405  found in single copy in      33 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -       Largest cog size: 15. Smallest cog size: 15\n",
      "\u001b[32mINFO\u001b[0m -       Analysis of current COG selection:\n",
      "\u001b[32mINFO\u001b[0m -                                555778 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                               1255043 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                263820 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                525897 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                273063 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                673860 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                               1121405 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                               1158165 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                933801 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                             224324999 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                224324 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                               1051632 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                713587 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                743299 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                637389 species present in     33 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -        33 COGs selected with at least 0 species out of 15\n",
      "\u001b[32mINFO\u001b[0m -        Average COG size 15.0/15.0 +- 0.0\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[34mCogSelectorTask\u001b[0m (15 species, MCL-COGs, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:41:57 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 31(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:41:59 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 29(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:01 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:03 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 27(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:05 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 25(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:07 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:09 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 23(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:11 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 21(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:13 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:15 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 19(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:17 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 17(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:19 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:22 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 15(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:24 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 13(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:26 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:28 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 11(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:30 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 9(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:32 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:34 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 7(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:36 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 5(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:38 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:40 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 3(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:42 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mwaiting for 2 cores\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 2 jobs. 2(R), 1(W). Cores usage: 4/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:44 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 1 jobs. 1(R), 0(W). Cores usage: 2/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:46 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[1;37;43mWRNG\u001b[0m -       Concatenating alignments\n",
      "\u001b[1;37;43mWRNG\u001b[0m -       Using aa concatenated alignment\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 33 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:48 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 1 jobs. 1(R), 0(W). Cores usage: 2/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:50 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 0 jobs. 1(R), 0(W). Cores usage: 2/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:52 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:54 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 0 jobs. 1(R), 0(W). Cores usage: 2/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:56 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 0 jobs. 1(R), 0(W). Cores usage: 2/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:42:58 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:43:00 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 0 jobs. 1(R), 0(W). Cores usage: 2/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:43:02 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 0 jobs. 1(R), 0(W). Cores usage: 2/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:43:04 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:43:06 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 0 jobs. 0(R), 0(W). Cores usage: 0/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:43:08 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 aa seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;33mLaunched\u001b[0m 0 jobs. 0(R), 0(W). Cores usage: 0/5\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Tue Sep  1 12:43:10 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[36mTreeMergeTask\u001b[0m (15 aa seqs, TreeMerger, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[36mTreeMergeTask\u001b[0m (15 aa seqs, TreeMerger, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  Assembling final tree...\n",
      "\u001b[32mINFO\u001b[0m -  Done thread \u001b[1;37;45mcog_all-alg_concat_default-fasttree_full\u001b[0m in 1 iteration(s)\n",
      "\u001b[32mINFO\u001b[0m -  Writing final tree for \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m\n",
      "   /home/huerta/etecourse/course/nb/03-building_concat_species_trees/data/sptree/cog_all-alg_concat_default-fasttree_full/all_prots.faa.final_tree.nw\n",
      "   /home/huerta/etecourse/course/nb/03-building_concat_species_trees/data/sptree/cog_all-alg_concat_default-fasttree_full/all_prots.faa.final_tree.nwx (newick extended)\n",
      "\u001b[32mINFO\u001b[0m -  Writing root node alignment \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m\n",
      "   /home/huerta/etecourse/course/nb/03-building_concat_species_trees/data/sptree/cog_all-alg_concat_default-fasttree_full/all_prots.faa.final_tree.fa\n",
      "\u001b[32mINFO\u001b[0m -  Done\n",
      "\u001b[32mINFO\u001b[0m -  Deleting temporal data...\n",
      "   ========================================================================\n",
      "         The following published software and/or methods were used.        \n",
      "               *** Please, do not forget to cite them! ***                 \n",
      "   ========================================================================\n",
      "   Huerta-Cepas J, Serra F, Bork P. ETE 3: Reconstruction, analysis and\n",
      "      visualization of phylogenomic data. Mol Biol Evol (2016) doi:\n",
      "      10.1093/molbev/msw046\n",
      "   Katoh K, Kuma K, Toh H, Miyata T. MAFFT version 5: improvement in\n",
      "      accuracy of multiple sequence alignment. Nucleic Acids Res. 2005 Jan\n",
      "      20;33(2):511-8.\n",
      "   Price MN, Dehal PS, Arkin AP. FastTree 2 - approximately maximum-\n",
      "      likelihood trees for large alignments. PLoS One. 2010 Mar\n",
      "      10;5(3):e9490.\n"
     ]
    }
   ],
   "source": [
    "# TASK: Execute ete build to infer a species tree out of the marker COGs\n",
    "!ete3 build -w mafft_default-none-none-none -m sptree_fasttree_all -o data/sptree \\\n",
    "           --cogs data/marker_cogs.tsv -a data/all_prots.faa --spname-delimiter . --clearall --cpu 5 --noimg\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You can also use codon alignments to infer the tree based on nucleotide rather than amino acid residues (useful when sequences are too similar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Toolchain path: /home/huerta/miniconda3/envs/eccb20/bin/ete3_apps/bin \n",
      "Toolchain version: unknown\n",
      "['mafft_default-none-none-none']\n",
      "['cog_all-alg_concat_default-fasttree_full']\n",
      "\n",
      "      --------------------------------------------------------------------------------\n",
      "                  ETE build (3.1.2) - reproducible phylogenetic workflows\n",
      "\n",
      "      Citation:\n",
      "\n",
      "       Huerta-Cepas J, Serra F and Bork P. ETE 3: Reconstruction, analysis and\n",
      "       visualization of phylogenomic data. Mol Biol Evol (2016)\n",
      "       doi:10.1093/molbev/msw046\n",
      "\n",
      "      (Note that a list of the external programs used to complete all necessary\n",
      "      computations will be shown after workflow execution. Those programs should\n",
      "      also be cited.)\n",
      "\n",
      "      --------------------------------------------------------------------------------\n",
      "      \n",
      "\u001b[32mINFO\u001b[0m -  Testing x86-64  portable applications...\n",
      "       clustalo: \u001b[32mOK\u001b[0m - 1.2.4\n",
      "      dialigntx: \u001b[32mOK\u001b[0m - This is DIALIGN-TX Version 1.0.2 - A Multiple Sequence alignment program.\n",
      "       fasttree: \u001b[32mOK\u001b[0m - FastTree Version 2.1.8 Double precision (No SSE3), OpenMP (2 threads)\n",
      "         iqtree: \u001b[32mOK\u001b[0m - IQ-TREE multicore version 1.5.5 for Linux 64-bit built Jun  2 2017\n",
      "         kalign: \u001b[32mOK\u001b[0m - Kalign version 2.04, Copyright (C) 2004, 2005, 2006 Timo Lassmann\n",
      "          mafft: \u001b[32mOK\u001b[0m - MAFFT v6.861b (2011/09/24)\n",
      "         muscle: \u001b[32mOK\u001b[0m - MUSCLE v3.8.31 by Robert C. Edgar\n",
      "          phyml: \u001b[32mOK\u001b[0m - . This is PhyML version 20160115.\n",
      "     pmodeltest: \u001b[32mOK\u001b[0m - pmodeltest.py v1.4\n",
      "          raxml: \u001b[32mOK\u001b[0m - This is RAxML version 8.2.11 released by Alexandros Stamatakis on June 2017.\n",
      " raxml-pthreads: \u001b[32mOK\u001b[0m - This is RAxML version 8.2.11 released by Alexandros Stamatakis on June 2017.\n",
      "         readal: \u001b[32mOK\u001b[0m - readAl v1.4.rev6 build[2012-02-02]\n",
      "         statal: \u001b[32mOK\u001b[0m - statAl v1.4.rev6 build[2012-02-02]\n",
      "        tcoffee: \u001b[32mOK\u001b[0m - PROGRAM: T-COFFEE Version_11.00.8cbe486 (2014-08-12 22:05:29 - Revision 8cbe486 - Build 477)\n",
      "         trimal: \u001b[32mOK\u001b[0m - trimAl v1.4.rev6 build[2012-02-02]\n",
      "\u001b[32mINFO\u001b[0m -  Starting ETE-build execution at Sun Aug 30 17:07:46 2020\n",
      "\u001b[32mINFO\u001b[0m -  Output directory /home/huerta/_Devel/etecourse/nb/03-building_concat_species_trees/data/sptree\n",
      "\u001b[32mINFO\u001b[0m -  Erasing all existing npr data...\n",
      "\u001b[1;37;43mWRNG\u001b[0m -  Using existing dir: /home/huerta/_Devel/etecourse/nb/03-building_concat_species_trees/data/sptree/db\n",
      "\u001b[1;37;43mWRNG\u001b[0m -  COG file restriction: 510 sequences from 15 species \n",
      "\u001b[32mINFO\u001b[0m -  Reading aa sequences from data/all_prots.faa...\n",
      "loaded:0000136 skipped:0009864 scanned:0010000 26.7%\n",
      "loaded:0000272 skipped:0019728 scanned:0020000 53.3%\n",
      "loaded:0000393 skipped:0029607 scanned:0030000 77.1%\n",
      "loaded:0000505 skipped:0039495 scanned:0040000 99.0%\n",
      "\n",
      "\n",
      "\u001b[32mINFO\u001b[0m -  Reading nt sequences from data/all_prots.fna...\n",
      "loaded:0000135 skipped:0009865 scanned:0010000 26.5%\n",
      "loaded:0000271 skipped:0019729 scanned:0020000 53.1%\n",
      "loaded:0000374 skipped:0029626 scanned:0030000 73.3%\n",
      "loaded:0000490 skipped:0039510 scanned:0040000 96.1%\n",
      "\n",
      "\n",
      "\u001b[1;37;43mWRNG\u001b[0m -  510 target sequences\n",
      "\u001b[32mINFO\u001b[0m -  ETE build starts now!\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:07:46 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[34mCogSelectorTask\u001b[0m (15 species, MCL-COGs, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -                     933801  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     273063  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     263820  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     673860  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                    1051632  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                  224324999  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     224324  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     525897  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     637389  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     743299  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     555778  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                    1158165  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                     713587  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                    1255043  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -                    1121405  found in single copy in      34 (97.1%) COGs \n",
      "\u001b[32mINFO\u001b[0m -       Largest cog size: 15. Smallest cog size: 15\n",
      "\u001b[32mINFO\u001b[0m -       Analysis of current COG selection:\n",
      "\u001b[32mINFO\u001b[0m -                                713587 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                             224324999 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                933801 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                224324 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                555778 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                273063 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                525897 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                263820 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                637389 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                743299 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                               1051632 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                               1121405 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                               1255043 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                               1158165 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -                                673860 species present in     34 COGs (100.0%)\n",
      "\u001b[32mINFO\u001b[0m -        34 COGs selected with at least 0 species out of 15\n",
      "\u001b[32mINFO\u001b[0m -        Average COG size 15.0/15.0 +- 0.0\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[34mCogSelectorTask\u001b[0m (15 species, MCL-COGs, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:07:48 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:07:51 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.66 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.66 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:07:53 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.61 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:07:55 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.57 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:07:57 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.69 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.64 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:07:59 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:01 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.72 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.69 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:03 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.64 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.67 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:05 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.68 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.69 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:07 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:09 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.65 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.63 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:11 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.71 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.66 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:13 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:15 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.66 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.59 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:17 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.66 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.65 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:19 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:21 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.70 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.63 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:23 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.68 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.51 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:26 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:28 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.75 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.60 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:30 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.55 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.58 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:32 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:34 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.60 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.59 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:36 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.65 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.69 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:38 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.52 >= 0.00\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:40 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -          \u001b[35mSwitching to codon alignment!\u001b[0m amino-acid sequence similarity: 0.62 >= 0.00\n",
      "\u001b[1;37;43mWRNG\u001b[0m -       Concatenating alignments\n",
      "\u001b[1;37;43mWRNG\u001b[0m -       Using nt concatenated alignment\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[31mConcatAlgTask\u001b[0m (15 species, 34 COGs, ConcatAlg, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:42 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:44 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:46 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:48 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:50 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:52 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:54 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:56 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:08:58 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[32mTreeTask\u001b[0m (15 nt seqs, FastTree, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Sun Aug 30 17:09:00 2020)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 15\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[36mTreeMergeTask\u001b[0m (15 nt seqs, TreeMerger, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[36mTreeMergeTask\u001b[0m (15 nt seqs, TreeMerger, /\u001b[1;37;40mcog_all-al...ttree_full\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  Assembling final tree...\n",
      "\u001b[32mINFO\u001b[0m -  Done thread \u001b[1;37;45mcog_all-alg_concat_default-fasttree_full\u001b[0m in 1 iteration(s)\n",
      "\u001b[32mINFO\u001b[0m -  Writing final tree for \u001b[1;37;40mcog_all-alg_concat_default-fasttree_full\u001b[0m\n",
      "   /home/huerta/_Devel/etecourse/nb/03-building_concat_species_trees/data/sptree/cog_all-alg_concat_default-fasttree_full/all_prots.faa.final_tree.nw\n",
      "   /home/huerta/_Devel/etecourse/nb/03-building_concat_species_trees/data/sptree/cog_all-alg_concat_default-fasttree_full/all_prots.faa.final_tree.nwx (newick extended)CPU times: user 1.67 ms, sys: 5.17 ms, total: 6.84 ms\n",
      "Wall time: 1min 16s\n"
     ]
    }
   ],
   "source": [
    "# TASK: Execute ete build to infer a species tree out of the marker COGs, using codon alignments and nt sequences\n",
    "!ete3 build -w mafft_default-none-none-none -m sptree_fasttree_all -o data/sptree \\\n",
    "           --cogs data/marker_cogs.tsv -a data/all_prots.faa -n data/all_prots.fna --nt-switch-threshold 0.0 \\\n",
    "           --spname-delimiter . --clearall --cpu 5 --noimg\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\r\n",
      "  File \"/home/huerta/miniconda3/envs/etecourse/bin/ete3\", line 8, in <module>\r\n",
      "    sys.exit(main())\r\n",
      "  File \"/home/huerta/miniconda3/envs/etecourse/lib/python3.6/site-packages/ete3/tools/ete.py\", line 95, in main\r\n",
      "    _main(sys.argv)\r\n",
      "  File \"/home/huerta/miniconda3/envs/etecourse/lib/python3.6/site-packages/ete3/tools/ete.py\", line 269, in _main\r\n",
      "    args.func(args)\r\n",
      "  File \"/home/huerta/miniconda3/envs/etecourse/lib/python3.6/site-packages/ete3/tools/ete_annotate.py\", line 61, in run\r\n",
      "    tree.annotate_ncbi_taxa(args.taxid_attr)\r\n",
      "  File \"/home/huerta/miniconda3/envs/etecourse/lib/python3.6/site-packages/ete3/phylo/phylotree.py\", line 798, in annotate_ncbi_taxa\r\n",
      "    return ncbi.annotate_tree(self, taxid_attr=taxid_attr, tax2name=tax2name, tax2track=tax2track, tax2rank=tax2rank)\r\n",
      "  File \"/home/huerta/miniconda3/envs/etecourse/lib/python3.6/site-packages/ete3/ncbi_taxonomy/ncbiquery.py\", line 545, in annotate_tree\r\n",
      "    lineage = tax2track[node_taxid],\r\n",
      "KeyError: 224324999\r\n"
     ]
    }
   ],
   "source": [
    "!cp \"data/sptree/cog_all-alg_concat_default-fasttree_full/all_prots.faa.final_tree.nw\" \"data/sptree.nw\" \n",
    "\n",
    "!ete3 annotate --ncbi -t \"data/sptree.nw\"| ete3 view --ncbi "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And, of course, you can load the species tree and operate with it"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0\n",
      "Cophenetic distance matrix:\n",
      " ([[0, 0.9269901, 0.9766983999999999, 1.0198972, 0.992823, 0.992823, 1.776664, 1.8348280000000001, 0.9181431, 1.0569824, 0.9746603, 1.7396190000000002, 1.0144897, 1.0255883, 1.8264490000000002], [0.9269901, 0, 0.8590059, 0.9022047000000001, 1.0524091, 1.0524091, 1.8362501, 1.8944141, 0.6481650000000001, 0.9392898999999999, 0.8569678000000001, 1.7992051, 0.8967972000000001, 0.9078958, 1.8860351], [0.9766983999999999, 0.8590059, 0, 0.2902328, 1.1021174, 1.1021174, 1.8859584000000003, 1.9441224, 0.8501588999999999, 0.56747, 0.6050761, 1.8489134000000003, 0.2848253, 0.6560041, 1.9357434], [1.0198972, 0.9022047000000001, 0.2902328, 0, 1.1453162, 1.1453162, 1.9291572000000001, 1.9873212000000002, 0.8933577, 0.6106688000000001, 0.6482749000000001, 1.8921122000000001, 0.0933881, 0.6992029000000001, 1.9789422], [0.992823, 1.0524091, 1.1021174, 1.1453162, 0, 0.0, 1.351439, 1.4096030000000002, 1.0435621, 1.1824013999999998, 1.1000793, 1.314394, 1.1399086999999999, 1.1510072999999998, 1.401224], [0.992823, 1.0524091, 1.1021174, 1.1453162, 0.0, 0, 1.351439, 1.4096030000000002, 1.0435621, 1.1824013999999998, 1.1000793, 1.314394, 1.1399086999999999, 1.1510072999999998, 1.401224], [1.776664, 1.8362501, 1.8859584000000003, 1.9291572000000001, 1.351439, 1.351439, 0, 1.070012, 1.8274031, 1.9662424, 1.8839203, 0.753939, 1.9237497000000001, 1.9348482999999999, 1.061633], [1.8348280000000001, 1.8944141, 1.9441224, 1.9873212000000002, 1.4096030000000002, 1.4096030000000002, 1.070012, 0, 1.8855671000000003, 2.0244064, 1.9420843000000003, 1.032967, 1.9819137000000002, 1.9930123000000002, 0.41210899999999995], [0.9181431, 0.6481650000000001, 0.8501588999999999, 0.8933577, 1.0435621, 1.0435621, 1.8274031, 1.8855671000000003, 0, 0.9304429000000001, 0.8481208, 1.7903581, 0.8879502, 0.8990488, 1.8771881], [1.0569824, 0.9392898999999999, 0.56747, 0.6106688000000001, 1.1824013999999998, 1.1824013999999998, 1.9662424, 2.0244064, 0.9304429000000001, 0, 0.6853601, 1.9291974, 0.6052613000000001, 0.7362881, 2.0160274], [0.9746603, 0.8569678000000001, 0.6050761, 0.6482749000000001, 1.1000793, 1.1000793, 1.8839203, 1.9420843000000003, 0.8481208, 0.6853601, 0, 1.8468753, 0.6428674000000001, 0.367772, 1.9337053], [1.7396190000000002, 1.7992051, 1.8489134000000003, 1.8921122000000001, 1.314394, 1.314394, 0.753939, 1.032967, 1.7903581, 1.9291974, 1.8468753, 0, 1.8867047000000001, 1.8978032999999999, 1.024588], [1.0144897, 0.8967972000000001, 0.2848253, 0.0933881, 1.1399086999999999, 1.1399086999999999, 1.9237497000000001, 1.9819137000000002, 0.8879502, 0.6052613000000001, 0.6428674000000001, 1.8867047000000001, 0, 0.6937954000000001, 1.9735347000000003], [1.0255883, 0.9078958, 0.6560041, 0.6992029000000001, 1.1510072999999998, 1.1510072999999998, 1.9348482999999999, 1.9930123000000002, 0.8990488, 0.7362881, 0.367772, 1.8978032999999999, 0.6937954000000001, 0, 1.9846332999999998], [1.8264490000000002, 1.8860351, 1.9357434, 1.9789422, 1.401224, 1.401224, 1.061633, 0.41210899999999995, 1.8771881, 2.0160274, 1.9337053, 1.024588, 1.9735347000000003, 1.9846332999999998, 0]], ['1051632', '1121405', '1158165', '1255043', '224324', '224324999', '263820', '273063', '525897', '555778', '637389', '673860', '713587', '743299', '933801'])\n"
     ]
    }
   ],
   "source": [
    "from ete3 import PhyloTree\n",
    "t = PhyloTree(\"data/sptree/cog_all-alg_concat_default-fasttree_full/all_prots.faa.final_tree.nw\")\n",
    "\n",
    "# Phylogenetic distance from our strain to the reference genome:\n",
    "print(t.get_distance(\"224324999\", \"224324\"))\n",
    "\n",
    "# Phylogenetic distance from all taxa\n",
    "print(\"Cophenetic distance matrix:\\n\",  t.cophenetic_matrix())"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
